Added minutes from TSC meeting 17 Jan 2019 (#324)
AcademySoftwareFoundation/openvdb: tsc/meetings/2019-01-17.md

Attendees: John M., *Jeff* L., *Nick* A., *Ken* M., *Peter* C., *Dan* B., Andrew P.

Additional Attendees: Bruce Chernia (Intel), Daniel Heckenberg (Animal Logic)

Agenda:

1) Confirmation of quorum
2) Selection of secretary
3) Update from TAC (Ken)
4) Update on Multi-Res grid from Autodesk (Ken)
5) Intel representative
6) JIT for OpenVDB (Nic)
7) Copyright notice (Dan)
8) Write permissions (Dan)
9) Release process (Dan)
10) Cleanup and improve repository (Ken)
11) Schedule next meeting

1) A quorum was confirmed.

2) Secretary - Nick Avramoussis.

3) OpenColourIO is accepted within the probationary stage. There's been interest form 3rd party
companies, including Nvidia, Intel and Autodesk.

Security and CI:

The issue of a security expert was discussed as something which should be addressed across all
ASWF projects. Jeff mentioned that the guidelines are vague for this role and that in the context
of VDB, this mainly relates to bug handling/invalid file reading. Ken mentioned the concern of
arbitrary metadata in regards to attacks. Andrew advised on static code analysis tools as part
of a CI build (lgtm, clang). Dan mentioned using clang tidy as a CI step which would be easier

> to solve on the CMake side to address CI best practices.

Reviewer: ```suggestion
to solve on the CMake side to address CII best practices.
```

CII stands for "Core Infrastructure Initiative" (https://bestpractices.coreinfrastructure.org/en)

The viability of Travis support was questioned. Previous limitations have been removed (Nov 2018)
which now allow for building on Ubuntu 16.04 for Houdini 17. Dan has a pull request which
optimises the Travis builds to help with timeouts, though it's agreed that it is not a sustainable
solution. It was agreed that the main thing holding us back from a Jenkins transition was the
CMake build.

CMake:

A fully supported CMake build through CI is becoming more of a priority. Current TSC members
do not have extensive experience with CMake. Nick mentioned his experience and usage within
DNEG production and will share example modification. It was agreed that other expertise would

> be ideal for deciding on a final structure. OpenColorIO (and others) have extensive CMake

Reviewer: I think the suggestion here is not that the CMake system within OpenColorIO would become standardized, just that the relative level of experience for CMake within the OpenColorIO project is greater than the OpenVDB project so perhaps this is an area where OpenColorIO could lead the way on establishing and adhering to CMake conventions and OpenVDB follow. 

(see the number of contributors here for example - https://github.com/imageworks/OpenColorIO/blob/master/CMakeLists.txt)

Author: Yeah this is what I mean't I'll re-word to be clearer

systems which could be standardized across the ASWF. Ken mentioned a concern if this was actioned

> due to the amount dependencies OpenVDB has. Ken mentioned that WETA have dedicated a lot of time

Reviewer: ```suggestion
due to the amount of dependencies OpenVDB has. Ken mentioned that WETA have dedicated a lot of time
```

5) Bruce Chernia (Intel) spoke on a desire to investigate OpenVDB integration with Embree and
OSPRay. These tools are exclusively focused on real time ray tracing on the CPU. Bruce mentioned
that they would be happy to contribute back findings to the OpenVDB project during their
investigations.

6) Nick spoke on the current state of OpenVDB AX. A stable V1.0.0 release is still planed and
will involve an incorporation into OpenVDB. This release will incorporate all items visible
on the OpenVDB AX roadmap, available within the AX repository on DNEGs github. Nick mentioned
that a summer release is most likely, with an integration to be performed before SIGGRAPH 2019.
Jeff and Nick discussed concerns in regards to Houdini integration, with a commitment being
made to support various VEX syntax.

SIGGRAPH:

Ken brought up the OpenVDB course at SIGGRAPH 2019. Ken will send around an email thread to
start a discussion on topics to be presented.

EMail Lists:

Jon brought up that TSC members should try where possible to use the aswf e-mail lists. Dan
mentioned that the private e-mail list is currently public. Jon to check and confirm a switch
to private.

Contributing File:

> A clause in the CONTRIBUTING.md file which currently declares a committers responsibility as

Reviewer: ```suggestion
A clause in the CONTRIBUTING.md file which currently declares a committer's responsibility as
```

'25% of one's time' is to be maintained but re-worded. The main desire is to ensure TSC members
can be held accountable if no work is apparent.

> 7) The OpenVDB copy-write notices need to be updated for 2019. A unanimous agreement was made to

Reviewer: ```suggestion
7) The OpenVDB copyright notices need to be updated for 2019. A unanimous agreement was made to
```

> switch to a SPDX license. This can be done independently of removing the copy-write dates.

Reviewer: ```suggestion
switch to a SPDX license. This can be done independently of removing the copyright dates.
```

> Jon recommended to add an additional copy-write with the dates removed and not remove the current

Reviewer: ```suggestion
Jon recommended to add an additional copyright with the dates removed and not remove the current
```

DreamWorks one.

8) Ken cannot be added as a reviewer onto current PRs and will check his write access status on
the repository. Currently no TSC members have admin access to the ASWF repo, required for setting
code review and pull request policy. It was agreed that Ken, Peter, Dan and Nick will be given
admin rights. Ken to submit this request to the LF. Dan to send an e-mail to the ASWF helpdesk
regarding JIRA watchers.

9) An official OpenVDB 6 announcement needs to be performed. Peter is the only TSC member with
access to the website. Peter suggested on a vote in the next TSC meeting on whether we continue
binary distributions.

10) ~Time

> 11) Our next TSC meeting is scheduled for Thursday Jan 14, 2019 11pm-12pm PST.

Reviewer: ```suggestion
11) Our next TSC meeting is scheduled for Thursday Jan 24, 2019 11am-12pm PST.
```
